% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={mitoGenome Assembly},
  pdfauthor={Ian G. Brennan},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\renewcommand{\linethickness}{0.05em}

\title{mitoGenome Assembly}
\author{Ian G. Brennan}
\date{14 January 2020}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{2}
\tableofcontents
}
\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\pagebreak

\hypertarget{install-the-software}{%
\section{Install the software}\label{install-the-software}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  MITObim\footnote{Hahn, C., Bachmann, L., Chevreux, B. Reconstructing
    mitochondrial genomes directly from genomic next-generation
    sequencing - a baiting and iterative mapping approach. Nucleic Acids
    Research 41:13. \url{doi:10.1093/nar/gkt371} (2013).}:
  \url{https://github.com/chrishah/MITObim}

  \begin{itemize}
  \tightlist
  \item
    make sure to follow install instructions, including dependecies:
    MIRA, PERL, etc.
  \end{itemize}
\item
  MUSCLE\footnote{Edgar, R.C. MUSCLE: multiple sequence alignment with
    high accuracy and high throughput. Nucleic Acids Research 32(5).
    \url{doi:10.1093/nar/gkh340} (2004).}:
  \url{https://www.drive5.com/muscle/}
\item
  MAFFT\footnote{Katoh, Standley. MAFFT multiple sequence alignment
    software version 7: improvements in performance and usability.
    Molecular Biology and Evolution 30:772-780 (2013).}:
  \url{https://mafft.cbrc.jp/alignment/software/}
\end{enumerate}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

If you're on a mac, downloading this repository will include MUSCLE and
MAFFT suited for a mac, so you can plug `n' play. If you're on a Windows
or Linux machine, you'll have to do this separately, and adjust the call
to the programs in `mitoAlign.R'. Sorry, I'm just too dumb.

\pagebreak

\hypertarget{multiple-mtgenome-assemblyalignment-using-mitobim-from-r}{%
\section{Multiple mtGenome assembly/alignment using MITObim from
R}\label{multiple-mtgenome-assemblyalignment-using-mitobim-from-r}}

\hypertarget{why-bother}{%
\subsection{Why Bother?}\label{why-bother}}

As a result of the exon capture process for Anchored Hybrid Enrichment
projects, we're getting a considerable amount of mitochondrial bycatch
in our raw reads. This mtDNA can provide a lot of information including
a separate phylogenetic history for the group of interest, or even past
introgression events. Also, ::\emph{free data}::.

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{file-preparation}{%
\subsection{File Preparation}\label{file-preparation}}

We'll use the MITObim pipeline, run from R to assemble our mtGenomes
against a reference (find one on GenBank, distantly related is OK).
We'll then pull out the final assembly for each sample and align them
against each other using MUSCLE.\\
Start by creating a directory to hold:

\begin{itemize}
\tightlist
\item
  the MITObim pipeline (`MITObim.pl')\\
\item
  the MIRA sequence assembler and mapper directory
  (`mira.4.0.2\ldots{}')\\
\item
  the MUSCLE executable (`muscle3.8.31\ldots{}')
\item
  the MAFFT folder (`mafft-mac')
\item
  a project-specific directory holding all your mtDNA reads for each
  sample in subdirectories (`Frogs'), and your reference mtGenome as a
  fasta file, labelled '{[}taxon{]}\_mtGenome.fasta'
\end{itemize}

Here's a quick schematic of what the structure should look like:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{/PATH\_TO\_PARENT\_DIRECTORY/mtGenomes}
\KeywordTok{|}\ExtensionTok{{-}{-}}\NormalTok{ MITObim.pl}
\KeywordTok{|}\ExtensionTok{{-}{-}}\NormalTok{ mira\_4.0.2\_darwin13.1.0\_x86\_64\_static}
\KeywordTok{|}   \KeywordTok{|}\ExtensionTok{{-}{-}}\NormalTok{ bin (et al.)}
\KeywordTok{|}\ExtensionTok{{-}{-}}\NormalTok{ muscle3.8.31\_i86darwin64}
\KeywordTok{|}\ExtensionTok{{-}{-}}\NormalTok{ bbmap}
\KeywordTok{|}   \KeywordTok{|}\ExtensionTok{{-}{-}}\NormalTok{ reformat.sh}
\KeywordTok{|}   \KeywordTok{|}\ExtensionTok{{-}{-}}\NormalTok{ et al.}
\KeywordTok{|}\ExtensionTok{{-}{-}}\NormalTok{ Frogs}
    \KeywordTok{|}\ExtensionTok{{-}{-}}\NormalTok{ [taxon]\_mtGenome.fasta (reference genome)}
    \KeywordTok{|}\ExtensionTok{{-}{-}}\NormalTok{ Taxon1}
    \KeywordTok{|}   \KeywordTok{|}\ExtensionTok{{-}{-}}\NormalTok{ Taxon1\_R1.fastq.gz}
    \KeywordTok{|}   \KeywordTok{|}\ExtensionTok{{-}{-}}\NormalTok{ Taxon1\_R2.fastq.gz}
    \KeywordTok{|}\ExtensionTok{{-}{-}}\NormalTok{ Taxon2}
        \KeywordTok{|}\ExtensionTok{{-}{-}}\NormalTok{ Taxon2\_R1.fastq.gz}
        \KeywordTok{|}\ExtensionTok{{-}{-}}\NormalTok{ Taxon2\_R2.fastq.gz}
\end{Highlighting}
\end{Shaded}

\pagebreak

\hypertarget{executing-the-code}{%
\subsection{Executing the Code}\label{executing-the-code}}

Open a new R script, and source the functions we'll need

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{source}\NormalTok{(}\StringTok{"/PATH\_TO\_CODE/mtGenome\_Assembly.R"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Then set your working directory to the project-specific directory (I've
called this directory `Frogs'). This holds a subdirectory for each
sample, and your reference mitogenome.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{setwd}\NormalTok{(}\StringTok{"/PATH\_TO\_DIR/mtGenomes/Frogs"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{mitoassemble}{%
\subsubsection{mitoAssemble}\label{mitoassemble}}

The first function (\emph{mitoAssemble}) assumes that in the directory
`Frogs', there are one or more subdirectories (1 per sample), with one
or more gzipped fastq files of the raw reads. For each sample
(subdirectory), it will:

\begin{itemize}
\tightlist
\item
  merge (concatenate) any *fastq.gz files together.\\
\item
  copy your reference genome (fasta file)\\
\item
  call MITObim and attempt to assemble the mtGenome\\
\item
  copy the assembled mtGenome to a directory of all assemblies
\end{itemize}

The function can be either run sequentially (one mitogenome assembled
after another), or in parallel (multiple mitogenomes assembled
simultaneously), and requires as input just a few things:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mitoAssemble}\NormalTok{(num.iter, reference.name, project.name,}
\NormalTok{             write.shell, ncores, }\DataTypeTok{combine =} \KeywordTok{c}\NormalTok{(}\StringTok{"cat"}\NormalTok{, }\StringTok{"bbmap"}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{itemize}
\tightlist
\item
  \emph{num.iter} is the number of assembly iterations MITObim should
  try before it times out and moves on to the next sample. I usually
  leave this set to 100, sometimes it takes 4 iterations, sometimes 60,
  hard to know.\\
\item
  \emph{reference.name} is the unique name of your reference genome
  fasta file which precedes '\_mtGenome.fasta'\\
\item
  \emph{project.name} is what you'd like the generated directories to be
  called (where the assemblies are stored)
\item
  \emph{write.shell} if you'd like the function to write a shell script
  so that you can run MITObim in parallel
\item
  \emph{ncores} designate the number of cores to use if you want to run
  in parallel
\item
  \emph{combine} tells the function whether to use \emph{cat} to combine
  raw read files, or \emph{bbmap} to create an interleaved file that can
  be used with the `--paired' flag in MITObim
\end{itemize}

If run sequentially (\emph{write.shell = FALSE}), the function
\emph{mitoAssemble} will spit out all the assemblies to a new directory,
and tell you where it is:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{Assembly}\NormalTok{(s) completed and saved to}\OperatorTok{:}\StringTok{ }\ErrorTok{/}\NormalTok{Users}\OperatorTok{/}\NormalTok{Ian}\OperatorTok{/}\NormalTok{MITObim}\OperatorTok{/}\NormalTok{Assa}\OperatorTok{/}\NormalTok{Assa\_mtGenomes}
\end{Highlighting}
\end{Shaded}

Depending on how many you need to assemble, this could take a while.

If run in parallel (\emph{write.shell = TRUE}), the function will spit
out a shell script for you to drop into your terminal (outside R
suggested), and provide instructions:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Your shell script }\ControlFlowTok{for}\NormalTok{ running MITObim }\ControlFlowTok{in}\NormalTok{ parallel is written to}\OperatorTok{:}
\ErrorTok{/}\NormalTok{Users}\OperatorTok{/}\NormalTok{Ian}\OperatorTok{/}\NormalTok{MITObim}\OperatorTok{/}\NormalTok{Assa}\OperatorTok{/}\NormalTok{Assa\_parallel.txt}
\NormalTok{Execute the command }\ControlFlowTok{in}\NormalTok{ parallel by copy}\OperatorTok{/}\NormalTok{paste to your terminal}\OperatorTok{:}
\NormalTok{parallel }\OperatorTok{{-}}\NormalTok{j }\DecValTok{14} \OperatorTok{:::}\ErrorTok{:}\StringTok{ }\ErrorTok{/}\NormalTok{Users}\OperatorTok{/}\NormalTok{Ian}\OperatorTok{/}\NormalTok{MITObim}\OperatorTok{/}\NormalTok{Assa}\OperatorTok{/}\NormalTok{Assa\_parallel.txt}
\end{Highlighting}
\end{Shaded}

This requires you have \emph{parallel} installed to your machine, which
you can easily do with \emph{\$ brew install parallel}, assuming you
have \emph{homebrew} installed (highly recommended).

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{mitocollate}{%
\subsubsection{mitoCollate}\label{mitocollate}}

If you have used the parallel option \emph{write.shell = T} for the
function \emph{mitoAssemble}, then you'll need to pull together the best
assembly for each sample into a single location. This function will:

\begin{itemize}
\tightlist
\item
  find each completed assembly
\item
  rename the assembly according to the sample name
\item
  rename the file according to the sample name
\item
  copy it to a folder which will hold all the assemblies
\end{itemize}

The function requires only the name of the project:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mitoCollate}\NormalTok{(project.name)}
\end{Highlighting}
\end{Shaded}

\begin{itemize}
\tightlist
\item
  \emph{project.name} is what you'd like the generated directories to be
  called (where the assemblies are stored)
\end{itemize}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{mitoalign}{%
\subsubsection{mitoAlign}\label{mitoalign}}

The next function \emph{mitoAlign} will:

\begin{itemize}
\tightlist
\item
  combine all assembled mtGenomes into a single fasta alignment

  \begin{itemize}
  \tightlist
  \item
    ({[}project.name{]}\_Assembly\_Alignment.fasta)\\
  \end{itemize}
\item
  align the assemblies using MUSCLE or MAFFT, into a single final
  alignment

  \begin{itemize}
  \tightlist
  \item
    ({[}project.name\_Aligned\_Assemblies.fasta{]})
  \end{itemize}
\end{itemize}

The function \emph{mitoAlign} will spit out the final alignment into
your new directory, and tell you where it is:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{your alignment of mtGenome assemblies is called}\OperatorTok{:}
\StringTok{  }\ErrorTok{/}\NormalTok{Users}\OperatorTok{/}\NormalTok{Ian}\OperatorTok{/}\NormalTok{MITObim}\OperatorTok{/}\NormalTok{Assa}\OperatorTok{/}\NormalTok{Assa\_mtGenomes}\OperatorTok{/}\NormalTok{Assa\_Aligned\_Assemblies.fasta}
\end{Highlighting}
\end{Shaded}

The function requires little information to do its job well. Bare
minimum it should be given the project name, and your choice of aligner
(``MUSCLE'' or ``MAFFT''). You can also designate the reference genome
for aligning purposes:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mitoAlign}\NormalTok{(project.name, }\StringTok{"MUSCLE"}\NormalTok{, reference.name)}
\end{Highlighting}
\end{Shaded}

\begin{itemize}
\tightlist
\item
  \emph{project.name} what you've been calling the project, so it can
  find the folder with all your mitoGenome assemblies.\\
\item
  \emph{reference.name} if you'd like to improve your multi-mitoGenome
  alignment, you can align them back to your reference genome. By
  default, this is NULL, to specify it, just give it the name you
  provided in the \emph{mitoAssemble} step.
\end{itemize}

At the moment, I'd probably suggest using the MAFFT aligner with the
reference option if possible. Aligning the assembled mitogenomes without
a reference can result in some poor behavior from poorly assembled
genomes.

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{mitocheck}{%
\subsubsection{mitoCheck}\label{mitocheck}}

The function \emph{mitoCheck} will:

\begin{itemize}
\tightlist
\item
  read in your alignment, and provide information about missing/gaps in
  your data\\
\item
  remove taxa with low amounts of sequence if you'd like
\end{itemize}

You just have to give it the alignment and a little info:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mitoCheck}\NormalTok{(project.name, alignment, }\DataTypeTok{count.gaps=}\NormalTok{T, }\DataTypeTok{missing.threshold=}\OtherTok{NULL}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{itemize}
\tightlist
\item
  \emph{project.name} is again, just what you've been calling the
  project so far (name of the directory)\\
\item
  \emph{alignment} is the name of the output .fasta file from the
  \emph{mitoAlign} step\\
\item
  \emph{count.gaps} is just a TRUE/FALSE statement about whether to
  count gaps as missing data\\
\item
  \emph{missing.threshold} if you want to remove taxa that exceed a
  certain percentage of missing data, indicate that here (0 \textless{}
  x \textless{} 1)
\end{itemize}

If you do decide to trim taxa based on missing data, the function will
produce a new alignment and tell you the name:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{your reduced alignment is now called}\OperatorTok{:}
\ErrorTok{/}\NormalTok{Users}\OperatorTok{/}\NormalTok{Ian}\OperatorTok{/}\NormalTok{MITObim}\OperatorTok{/}\NormalTok{Assa}\OperatorTok{/}\NormalTok{Assa\_mtGenomes}\OperatorTok{/}\NormalTok{Assa\_Aligned\_Assemblies\_Reduced.fasta}
\end{Highlighting}
\end{Shaded}

I thought about using a different metric such as base
composition/content, but I haven't thought about it enough to implement
it well. Instead, it always identifies the outgroups as being funky, so
I've just stuck with missing data.

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{mitochop}{%
\subsubsection{mitoChop}\label{mitochop}}

The last function \emph{mitoChop} will:

\begin{itemize}
\tightlist
\item
  take the whole mitoGenome alignment and split it up by locus\\
\item
  this function requires the R packages \emph{ape} and \emph{seqinr} to
  read and write the files.
\end{itemize}

The function will kick out the alignment for each mitochondrial locus
(CDS/gene, rRNA, tRNA, \emph{et al}.) separately into a new folder:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{your separated mitochondrial loci alignments are }\ControlFlowTok{in}\NormalTok{ a folder called}\OperatorTok{::}
\StringTok{ }\ErrorTok{/}\NormalTok{Users}\OperatorTok{/}\NormalTok{Ian}\OperatorTok{/}\NormalTok{MITObim}\OperatorTok{/}\NormalTok{Assa}\OperatorTok{/}\NormalTok{Assa\_mtGenomes}\OperatorTok{/}\NormalTok{Assa\_mitoLoci}
\end{Highlighting}
\end{Shaded}

You just have to provide a character set file as a .CSV so it knows
where to cut it up

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mitoChop}\NormalTok{(project.name, alignment, }\DataTypeTok{character.sets=}\OtherTok{NULL}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{itemize}
\tightlist
\item
  \emph{project.name} is again, just what you've been calling the
  project so far (name of the directory)\\
\item
  \emph{alignment} the name of the output .fasta file from the
  \emph{mitoAlign} step\\
\item
  \emph{character.sets} is the .CSV file with three columns ``Name'',
  ``Minimum'', and ``Maximum''. These correspond to the locus name, the
  starting position (\emph{Minimum}), and final position
  (\emph{Maximum}) for each locus. An example is included in the
  respository (``mitoGenome\_Annotations\_Example.csv'').
\end{itemize}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

There's probably a bunch of other slick things I could do after this,
like have it plot the individual assembly lengths, but I'm not sure it's
worth it at the moment. Let me know if there's something specific you're
looking for though.\\
Good luck!

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\end{document}
